src/include/cpu/rpp_cpu_common.hpp

```diff
@@ -82,6 +83,53 @@ inline int power_function(int a, int b)
     return product;
 }
 
+inline void saturate_pixel(Rpp32f pixel, Rpp8u* dst) {
+    *dst = RPPPIXELCHECK(pixel);
+}
+
+inline void saturate_pixel(Rpp32f pixel, Rpp8s* dst) {
+    *dst = RPPPIXELCHECKI8(pixel);
+}
+
+inline void saturate_pixel(Rpp32f pixel, Rpp32f* dst) {
+    *dst = (Rpp32f)pixel;
+}
+
+inline void saturate_pixel(Rpp32f pixel, Rpp16f* dst) {
+    *dst = (Rpp16f)pixel;
+}
+
+inline float sinc(float x) {
+    x *= M_PI;
+    if (std::abs(x) < 1e-5f)
+    return 1.0f - x * x * (1.0f / 6);  // remove singularity by using Taylor expansion
+    return std::sin(x) / x;
+}
+
+inline void CalculateLanczosCoefficients(float* coeffs, float x, int a)
```

Comment:
This one needs to be optimized too. If 'a' is a pre-defined constant as defined in L1137, and you only need [a, 2 * a, 1 / a, a - 1] inside this functionality, please compute them in an array say aParams[4] or lanczosParams[4] and reuse them here. Currently, these 4 constant values are being computed width*height times for every single function call